### ISSUE TITLE
Guake does not appear in the right position when activated in Gnome 3.18 Wayland

### ISSUE THREAD

**Issue by hellslinger** [Guake/guake#806]
Pressing F12 causes guake to appear in the wrong place. Sometimes it appears in the middle of the two screens so that it spans the center of them, and sometimes it appears the wrong size, far to the left. I notice that when it first appears in the very top left of my leftmost screen, it is too small, as if it is being sized for my right screen which is far smaller than my left screen.

**Comment by Stibbons** [comment]
can you check if the option to place guake on the screen on your mouse isn't activated?

**Comment by hellslinger** [comment]
"Appear on mouse display" is enabled, however it works correctly in non-wayland gnome.

**Comment by stanfieldr** [comment]
I believe there is two issues here. The effect you're were describing about where it starts on one monitor and moves to the other works now. There's another problem I discovered while debugging this though. Guake uses the mouse position to determine which screen to popup on, but this doesn't seem to work in wayland. Apparently the mouse coordinates only update if you hover over gtk2 apps, not gtk3 or the desktop. See video:
[Screencast_11-23-2016_04:09:38 AM.zip](https://github.com/Guake/guake/files/608494/Screencast_11-23-2016_04.09.38.AM.zip)

But hey, at least we fixed part of the issue :D
